test/view_component/slotable_v2_test.rb

```diff
@@ -272,6 +272,17 @@ def test_slot_with_block_content
     assert_selector("p", text: "Footer part 2")
   end
 
+  def test_slot_with_block_content_returning_nil
+    component = SlotsV2Component.new
+    slot = component.title nil
+    slot.__vc_content_block = Proc.new { nil }
+    # Render the component so that the slot has a view_context
+    render_inline component
+
+    # No matter what, the slot shouldn't return nil from `to_s`
+    assert_not_nil(slot.to_s)
+  end
```

Comment:
This test _feels_ a little too coupled to implementation details. Might we instead add a case to `integration_test.rb` that mimics the failure mode in an actual slim file?